File: #1027.longest-arithmetic-subsequence.cpp
class Solution {
public:
    int longestArithSeqLength(vector<int>& nums) {

        int size = nums.size();
        memset(dp , 0 , sizeof(dp));
        int maxlen = 1;
        for(int i = 1 ; i < size ;i++) {
            for (int j = 0; j < i; j++) {
                int diff = nums[i] - nums[j] + 500;
                dp[i][diff] = max(dp[i][diff] , 1 + dp[j][diff]);
                maxlen = std::max(maxlen , dp[i][diff]);
            }
        }
        return maxlen + 1;
    }

private:
    int dp[1001][1001];
};

File: #1027.longest-arithmetic-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestArithSeqLength = function(nums) {

    let length = nums.length;
    let result = 1;
    for (let i = 0; i < length; i++) {
        for (let j = i + 1; j < length; j++) {
            let currDiff = nums[j] - nums[i];
            let next = nums[j] + currDiff;
            let count = 1;
            for (let k = j + 1; k < length; k++) {
                if (next === nums[k]) {
                    next = nums[k] + currDiff;
                    count++;
                }
            }
            result = Math.max(result, count + 1);
        }
    }
    return result;
};
